src/librustc_save_analysis/dump_visitor.rs

```diff
@@ -467,8 +482,13 @@ impl<'l, 'tcx: 'l, 'll, O: DumpOutput + 'll> DumpVisitor<'l, 'tcx, 'll, O> {
                 |v| v.process_formals(&decl.inputs, &fn_data.qualname),
             );
             self.process_generic_params(ty_params, item.span, &fn_data.qualname, item.id);
-            self.dumper
-                .dump_def(item.vis == ast::Visibility::Public, fn_data);
+            self.dumper.dump_def(
+                &Access {
+                    public: item.vis == ast::Visibility::Public,
+                    reachable: self.save_ctxt.analysis.access_levels.is_reachable(item.id),
+                },
```

Comment:
This code looks kinda boilerplatey - could you DRY it out by using a function or macro?

Reply:
"This code" = initialising `Access`